metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1058,15 +1070,20 @@ class MetalsLanguageServer(
 
   @JsonNotification("textDocument/didOpen")
   def didOpen(params: DidOpenTextDocumentParams): CompletableFuture[Unit] = {
-    val path = params.getTextDocument.getUri.toAbsolutePath
+    val localURI = uriMapper.convertToLocal(params.getTextDocument.getUri)
+    val path = localURI.toAbsolutePath
     // In some cases like peeking definition didOpen might be followed up by close
     // and we would lose the notion of the focused document
     focusedDocument.foreach(recentlyFocusedFiles.add)
     focusedDocument = Some(path)
     recentlyOpenedFiles.add(path)
 
     // Update md5 fingerprint from file contents on disk
-    fingerprints.add(path, FileIO.slurp(path, charset))
+    fingerprints.add(
+      path,
+      if (path.isJarFileSystem) params.getTextDocument.getText
```

Comment:
We shouldn't need to add fingerprints for jarFileSystem. There are really only useful if the file can change,

Reply:
I agree except I think fingerprints are used to get the text of the file in `SyntheticsDecorationProvider#enrichWithText` but I would like that to change so the text is passed in from the LSP message.

Reply:
So do you save it in the fingerprints to avoid the complexity of reading it from the file system?

Reply:
When I came to look at auto-decompiling class files using CFR and realised the decompile took a few seconds for every time that file is read from disk, I started tracing how often Metals uses `FileIO.slurp` and it seems to use it a lot and, to me, that seems unnecessary since all the LSP messages send us the file contents.  I see that fingerprinting is used to go back to the last successfully compiled version but I don't get why it's used in `SyntheticsDecorationProvider#enrichWithText` instead of just using the contents passed by LSP and I don't get why it's used in many other places e.g. InteractiveSemanticDB, when the text is readily available.  I doubt this has been an issue with any files present on the file system but when you have to decompile the `class` file then that has to be wired in and it still takes time so why do it at all?
But yes - mostly because of the additional decompile phase needed when slurping a class file.

Reply:
>  I see that fingerprinting is used to go back to the last successfully compiled version but I don't get why it's used in

We only have semanticdb for the last compiled state, so whenever we use it we try to check for the differences between the current version and the compiled one to calculate adjustments to positions. 

> But yes - mostly because of the additional decompile phase needed when slurping a class file.

I think that should be fine then, but for that case ideally we would use buffers, which contain all the currently opened files. Fingerprints are also saved currently in between sessions and there is no need to save them if they don't really change.